src/main/java/frc/robot/util/Vision.java

```diff
@@ -0,0 +1,176 @@
+/*----------------------------------------------------------------------------*/
+
+package frc.robot.util;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import org.photonvision.PhotonCamera;
+import org.photonvision.PhotonUtils;
+import org.photonvision.RobotPoseEstimator;
+import org.photonvision.RobotPoseEstimator.PoseStrategy;
+import org.photonvision.targeting.PhotonPipelineResult;
+import org.photonvision.targeting.PhotonTrackedTarget;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.math.MatBuilder;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.Pair;
+import edu.wpi.first.math.estimator.DifferentialDrivePoseEstimator;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Transform3d;
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.constants.Constants;
+import frc.robot.constants.FieldConstants;
+import frc.robot.constants.VisionConstants;
+
+
+
+public class Vision {
+  private static RobotPoseEstimator robotPoseEstimator;
+  // private static RobotPoseEstimator robotPoseEstimator2;
+  private static AprilTagFieldLayout aprilTagFieldLayout;
+ 
+
+  /**
+   * Calls the other setup function
+   */
+  public static void setup() {
+    setup(new PhotonCamera(VisionConstants.kCameraName1), new PhotonCamera(VisionConstants.kCameraName2));
+  }
+
+  /**
+   * Sets up the cameras and pose estimator
+   * @param camera1 The first camera
+   * @param camera2 The second camera
+   */
+  public static void setup(PhotonCamera camera1, PhotonCamera camera2) {
+    ArrayList<Pair<PhotonCamera, Transform3d>> camList;
+    if(VisionConstants.k2Cameras){
+      camList = new ArrayList<Pair<PhotonCamera, Transform3d>>(List.of(
+        new Pair<PhotonCamera, Transform3d>(camera1, VisionConstants.kRobotToCamera1),
+        new Pair<PhotonCamera, Transform3d>(camera2, VisionConstants.kRobotToCamera2)
+      ));
+    }else{
+      camList = new ArrayList<Pair<PhotonCamera, Transform3d>>(List.of(
+        new Pair<PhotonCamera, Transform3d>(camera1, VisionConstants.kRobotToCamera1)
+      ));
+    }
+    getTagFieldLayout();
+    aprilTagFieldLayout.setOrigin(AprilTagFieldLayout.OriginPosition.kBlueAllianceWallRightSide);
+    
+    robotPoseEstimator = new RobotPoseEstimator(aprilTagFieldLayout, PoseStrategy.CLOSEST_TO_REFERENCE_POSE, camList);
+
+    if(VisionConstants.k2Cameras){
```

Comment:
consider removing this

Reply:
If we want to remove one camera for any reason, we only need to change it in the constants file. There is also no reason to remove this.